topicgenerator/src/main/java/io/dockstore/topicgenerator/helper/StringFilter.java

```diff
@@ -0,0 +1,33 @@
+package io.dockstore.topicgenerator.helper;
+
+import java.util.Set;
+import org.apache.commons.lang3.StringUtils;
+
+public interface StringFilter {
+
+    /**
+     * @param topic
+     * @return true on failure  (i.e. should be filtered out and is caught by the filter)
+     */
+    boolean assessTopic(String topic);
+
+
+    /**
+     * Utility method for screening topics against sets of offensive words.
+     *
+     * @param topic
+     * @param offensiveWords
+     * @return
+     */
+    default boolean assessTopic(String topic, Set<String> offensiveWords) {
```

Comment:
A note on the API design:

The `StringFilter` interface appears to have a singular purpose: to determine if strings are "bad" or not.  This method is a utility method, which helps an implementation check a specified string against a list of words.

However, all `StringFilter` implementations won't necessarily match against a list of words.  They might use AI, or regexps, or some other mechanism to determine if the string is "bad".  Those implementations don't use a list of words, so for them, this helper method doesn't make any sense and will be confusing if it appears in their class signatures.

A cleaner and more typical class/interface hierarchy would be something like (for brevity, I've omitted the access keywords (`public`/`private`/etc):
```
interface StringFilter {
    boolean assessTopic(String topic);
}

class WordsStringFilter implements StringFilter {
    Set<String> words;
    WordsStringFilter(Set<String> words) {
        this.words = words;
    }
    boolean assessTopic(String topic) {
        // check topic against words field
        [...]
    }
}

class ChuckNorrisStringFilter extends WordsStringFilter {
     ChuckNorrisStringFilter() {
          super(readChuckNorrisWords());
     }
     static Set<String> readChuckNorrisWords() {
          [...]
     }
}
```
Essentially, this helper method has moved to the `WordsStringFilter` class, and all subclasses inherit it, preventing it from clouding the signature of other `StringFilter` classes that don't deal in lists of words.  If more word-oriented helpers are necessary, they can be added to `WordsStringFilter`, too.

Reply:
The suggested solution also would allow you to create a master word list (sourced from multiple sources) and simply instantiate a `WordsStringFilter` that uses it, with a minimum of additional code...